calico_containers/tests/unit/datastore_test.py

```diff
@@ -1211,6 +1212,136 @@ def test_get_default_node_as_no_config(self):
 
         assert_equal(self.datastore.get_default_node_as(), 64511)
 
+    def test_get_hostnames(self):
+        """
+        Test get_hostnames()
+        """
+        def mock_read(path):
```

Comment:
This isn't a very realistic read, since the host path in a live system will include all kinds of shit like containers & endpoints.  I'd like to test that the code deals with that.